ui/app/components/namespace-picker.js

```diff
@@ -3,168 +3,79 @@
-        // replace slash with dots
-        .replace(/\/+/g, '.')
-    );
-  },
-
-  // an array that keeps track of what additional panels to render
-  // on the menu stack
-  // if you're in  'foo/bar/baz',
-  // this array will be: ['foo', 'foo.bar', 'foo.bar.baz']
-  // the template then iterates over this, and does  Ember.get(namespaceTree, leaf)
-  // to render the nodes of each leaf
-
-  // gets set as  'lastMenuLeaves' in the ember concurrency task above
-  menuLeaves: computed('namespacePath', 'namespaceTree', 'pathToLeaf', function () {
-    let ns = this.namespacePath;
-    ns = (ns || '').replace(/^\//, '');
-    let leaves = ancestorKeysForKey(ns);
-    leaves.push(ns);
-    leaves = this.maybeAddRoot(leaves);
-
-    leaves = leaves.map(this.pathToLeaf);
-    return leaves;
-  }),
-
-  // the nodes at the root of the namespace tree
-  // these will get rendered as the bottom layer
-  rootLeaves: computed('namespaceTree', function () {
-    const tree = this.namespaceTree;
-    const leaves = Object.keys(tree);
-    return leaves;
-  }),
-
-  currentLeaf: alias('lastMenuLeaves.lastObject'),
-  canAccessMultipleNamespaces: gt('accessibleNamespaces.length', 1),
-  isUserRootNamespace: computed('auth.authData.userRootNamespace', 'namespacePath', function () {
-    return this.auth.authData.userRootNamespace === this.namespacePath;
-  }),
-
-  namespaceDisplay: computed('namespacePath', 'accessibleNamespaces', 'accessibleNamespaces.[]', function () {
-    const namespace = this.namespacePath;
-    if (!namespace) {
-      return 'root';
-    }
-    const parts = namespace?.split('/');
-    return parts[parts.length - 1];
-  }),
+/**
+ * @module ApplicationComponent component is used to display a dropdown listing all namespaces that the current user has access to.
+ *  The user can select a namespace from the dropdown to navigate directly to that namespace.
+ *  The "Manage" button directs the user to the namespace management page.
+ *  The "Refresh List" button refrehes the list of namespaces in the dropdown.
+ *
+ * @example
+ * <NamespacePicker class="hds-side-nav-hide-when-minimized" />
+ */
 
-  actions: {
-    refreshNamespaceList() {
-      this.namespaceService.findNamespacesForUser.perform();
-    },
-  },
-});
+export default class ApplicationComponent extends Component {
+  @service namespace;
+
+  @tracked selected = {};
+  @tracked options = [];
+
+  constructor() {
+    super(...arguments);
+    this.loadOptions();
+  }
+
+  #matchesPath(option, currentNamespace) {
+    // TODO: Revisit. A hardcoded check for "path" & "/path" seems hacky, but it fixes a breaking test:
+    //  "Acceptance | Enterprise | namespaces: it shows nested namespaces if you log in with a namespace starting with a /"
+    //  My assumption is that namespace shouldn't start with a "/", but is this a HVD thing? or is the test outdated?
+    return option?.path === currentNamespace?.path || `/${option?.path}` === currentNamespace?.path;
```

Comment:
Note: added TODO to revisit this, I don't like this solution and want to dig deeper to understand if the referenced test is still relevant.